boot/freeldr/freeldr/arch/i386/multiboot.S

```diff
@@ -173,23 +173,33 @@ mb3:
     mov ss, ax
 
     /* Jump to relocated code */
-    ljmp HEX(08), mb4
+    .byte HEX(EA)
+    .long mb4
+    .word HEX(08)
 
 mbfail:
     int 3
 mbstop:
     jmp short mbstop  /* We should never get here */
-
-.code16
 mb4:
     /* Disable protected mode */
+#ifdef _M_IX86
     mov eax, cr0
+#else
+    mov rax, cr0
+#endif
+    .byte HEX(66)
```

Comment:
I suggest to use `data32` prefix, like done here: https://git.reactos.org/?p=reactos.git;a=blob;f=hal/halx86/smp/i386/apentry.S;hb=08a68340750adffc7912e06fc761d762a8a9c22e#l33

Reply:
the data32 prefix doesn't compile in Linux Amd64 GCC and Clang.

Reply:
i used the data32prefix instead because it's compile in all the compilers bots.